scripts/popup.js

```diff
@@ -1,29 +1,95 @@
 function save_now_function(){
-	var wb_url = "https://web.archive.org/save/";
-	chrome.runtime.sendMessage({message: "openurl", wayback_url: wb_url, method:'save' }, function(response) {
-	});
+    var wb_url = "https://web.archive.org/save/";
+    chrome.runtime.sendMessage({message: "openurl", wayback_url: wb_url, method:'save' }, function(response) {
+    });
 }
 
 function recent_capture_function(){
-	var wb_url = "https://web.archive.org/web/2/";
-	chrome.runtime.sendMessage({message: "openurl", wayback_url: wb_url, method:'recent' }, function(response) {
-	});
+    var wb_url = "https://web.archive.org/web/2/";
+    chrome.runtime.sendMessage({message: "openurl", wayback_url: wb_url, method:'recent' }, function(response) {
+    });
 }
 
 function first_capture_function(){
-	var wb_url = "https://web.archive.org/web/0/";
-	chrome.runtime.sendMessage({message: "openurl", wayback_url: wb_url, method:'first' }, function(response) {
-	});
+    var wb_url = "https://web.archive.org/web/0/";
+    chrome.runtime.sendMessage({message: "openurl", wayback_url: wb_url, method:'first' }, function(response) {
+    });       
 }
-
 function view_all_function(){
-	var pattern = /https:\/\/web\.archive\.org\/web\/(.+?)\//g;
-	url = document.location.href.replace(pattern, "");
-	open_url = "https://web.archive.org/web/*/"+encodeURI(url);
-	document.location.href = open_url;
+    var pattern = /https:\/\/web\.archive\.org\/web\/(.+?)\//g;
+    url = document.location.href.replace(pattern, "");
+    open_url = "https://web.archive.org/web/*/"+encodeURI(url);
+    document.location.href = open_url;
 }
+function fb_share(search_url){    
+    var fburl="https://www.facebook.com/sharer/sharer.php?u=";
+    chrome.tabs.create({url:fburl+search_url});
+}
+function twitter_share(search_url){    
+    var tweeturl="https://twitter.com/intent/tweet?text=";
+    chrome.tabs.create({url:tweeturl+search_url});
+}
+function google_share(search_url){    
+    var googleurl="https://plus.google.com/share?url=";
+    chrome.tabs.create({url:googleurl+search_url});
+}
+window.onload=function start()
+{
+    document.getElementById("form").addEventListener("keydown",function(e)
+    {
+        if(e.keyCode===13)
+        {
+            var http=new XMLHttpRequest();                                          
+            var take_url=document.getElementById("form").value;
+            var new_url="http://archive.org/wayback/available?url="+take_url;
+            http.open("GET",new_url,true);
+            http.send(null);
+            http.onload=function()
+            {
+                var data=JSON.parse(http.response);
+                if(typeof data.archived_snapshots.closest !=="undefined")
+                {
+                    var timestamp_date=data.archived_snapshots.closest.timestamp;
```

Comment:
Yes, it would work in this case but it should be `if(data.archived_snapshots.closest)`.
But I generally avoid checking it that way as it checks for truthy values. (https://www.sitepoint.com/javascript-truthy-falsy/). e.g.

`var b = 10;
if(b) console.log(“hi”)` // Prints hi

`var b = 0;
if(b) console.log(“hi”)`  // does not print

`var b = “”
if(b) console.log(“hi”)`  // does not print

To avoid this inconsistency I normally do a type-check. But yes in this case your suggestion works. But I still stick to my style of evaluation unless there’s a forced standard.